src/PdoFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Starlit Db.
+ *
+ * @copyright Copyright (c) 2019 Starweb AB
+ * @license   BSD 3-Clause
+ */
+
+namespace Starlit\Db;
+
+use PDO;
+
+class PdoFactory implements PdoFactoryInterface
+{
+    public function createPdo(string $dsn, string $username = null, string $password = null, array $options = []): PDO
+    {
```

Comment:
Shouldn't we also add the previous `$defaultPdoOptions` here?
Not too good if they are omited since that would affect performance and introduce issues (both with `ERRMODE_EXCEPTION` and `FETCH_ASSOC` for example)